README.md

```diff
@@ -119,44 +113,83 @@ type OnRetry func(n uint, err error)
 
 Function signature of OnRetry function n = count of tries
 
-#### type RetryOpts
+#### type Option
 
 ```go
-type RetryOpts struct {
-}
+type Option func(*config)
 ```
 
-Struct for configure retry tries - count of tries delay - waiting time units -
-waiting time unit (for tests purpose)
+Option represents an option for retry.
 
-#### func  NewRetryOpts
+#### func  Delay
 
 ```go
-func NewRetryOpts() RetryOpts
+func Delay(delay time.Duration) Option
 ```
-Create new RetryOpts struct with default values default tries are 10 default
-delay are 1e5 default units are microsecond
+Delay set delay between retry default are 1e5 units
 
-#### func (RetryOpts) Delay
+#### func  OnRetryFunction
 
 ```go
-func (opts RetryOpts) Delay(delay time.Duration) RetryOpts
+func OnRetryFunction(onRetryFunction OnRetry) Option
 ```
-Delay setter
+OnRetryFunction function callback are called each retry
 
-#### func (RetryOpts) Tries
+log each retry example:
+
+    retry.Do(
+    	func() error {
+    		return errors.New("some error")
+    	},
+    	retry.OnRetryFunction(func(n unit, err error) {
+    		log.Printf("#%d: %s\n", n, err)
+    	}),
+    )
+
+#### func  RetryIfFunction
 
 ```go
-func (opts RetryOpts) Tries(tries uint) RetryOpts
+func RetryIfFunction(retryIfFunction RetryIfFunc) Option
 ```
-Tries setter
+RetryIfFunction controls whether a retry should be attempted after an error
```

Comment:
So, assuming that maximum number of attempts still applies, is the main use case for this to allow certain kinds of errors to immediately bubble up (i.e. do not retry for these errors)?

Reply:
yes, it is main use case...

or is possible to do retry only if error is some special kind of error - but it's same, only negative approach